tests/test_flagging_grouping.py

```diff
@@ -0,0 +1,106 @@
+"""Tests for grouping and flagging"""
+import logging
+import os
+import unittest
+import pytest
+import copy
+
+import astropy.units as u
+import numpy as np
+
+import pint.models.model_builder as mb
+import pint.toa as toa
+from pint.residuals import Residuals
+from pinttestdata import datadir
+from pint.models import parameter as p
+from pint.models import PhaseJump
+
+
+class SimpleSetup:
+    def __init__(self, par, tim):
+        self.par = par
+        self.tim = tim
+        self.m = mb.get_model(self.par)
+        self.t = toa.get_TOAs(
+            self.tim, ephem="DE405", planets=False, include_bipm=False
+        )
+
+
+@pytest.fixture
+def setup_NGC6440E():
+    os.chdir(datadir)
```

Comment:
Please do not change directories in a test. This is modifying global state and can break things if a subset of tests are run. Just use path names `os.path.join(datadir, "file.par")`.